src/main/scala/com/agar/core/region/Region.scala

```diff
@@ -1,88 +1,253 @@
+      case (false, true) =>
+        bridge ! Virtual(RegisterPlayer(player, state))
+      case (true, false) =>
+        if (isInRegion(state.position)) {
+          bridge ! Virtual(Destroy(player))
+        } else {
+          context.stop(player)
+          bridge ! Virtual(CreatePlayer(state))
+        }
+      case (false, false) =>
+        // Small frontier isn't it?
+        if (!isInRegion(state.position)) {
+          context.stop(player)
+          bridge ! Virtual(CreatePlayer(state))
+        }
+    }
+
+    inFrontier
+  }
+
+  private def manageVirtualEnergy(energy: ActorRef, state: EnergyState): Boolean = {
+    val inFrontier = isInFrontier(state.position)
+
+    if (inFrontier) {
+      bridge ! Virtual(RegisterEnergy(energy, state))
+    }
+
+    inFrontier
+  }
+
+  def initializeEntities(nbOfPlayer: Int, nbOfStartingEnergy: Int): Unit = {
     val r = new scala.util.Random
 
-    this.players = (0 until nbOfPlayer)
-      .map { n =>
-        val position = Vector2d(r.nextInt(width), r.nextInt(height))
-        val player = createNewPlayer(position)
-        player -> PlayerState(position, WEIGHT_AT_START, DEFAULT_VELOCITY)
+    this.players = players ++ (0 until nbOfPlayer)
+      .map { _ =>
+        val position: Vector2d = generatePosition(r)
+        val player = createNewPlayer(position, WEIGHT_AT_START)
+        val state = PlayerState(position, WEIGHT_AT_START, DEFAULT_VELOCITY)
+        val virtual = manageVirtualPlayer(player, state)
+
+        player -> state.virtual(virtual)
       }.toMap
 
-    this.energies = (0 until nbOfStartingEnergy)
+    this.energies = energies ++ (0 until nbOfStartingEnergy)
       .map { _ =>
-        val position = Vector2d(r.nextInt(width), r.nextInt(height))
+        val position: Vector2d = generatePosition(r)
         val powerOfTheEnergy = 1 + r.nextInt(MAX_ENERGY_VALUE)
         val energy = createNewEnergy(powerOfTheEnergy)
-        energy -> EnergyState(position, powerOfTheEnergy)
+        val state = EnergyState(position, powerOfTheEnergy)
+        val virtual = manageVirtualEnergy(energy, state)
+
+        energy -> state.virtual(virtual)
       }.toMap
   }
 
-  private def createNewPlayer(position: Vector2d): ActorRef = {
-    context.actorOf(Player.props(position, WEIGHT_AT_START)(self))
+  private def generatePosition(r: Random) = {
+    val partition = 7000
+
+    Vector2d(r.nextInt(partition) * width / partition, r.nextInt(partition) * height / partition)
+  }
+
+  private def createNewPlayer(position: Vector2d, weight: Int): ActorRef = {
+    context.actorOf(Player.props(position, weight)(self))
   }
 
   private def createNewEnergy(valueOfEnergy: Int): ActorRef = {
-    context.actorOf(Energy.props(valueOfEnergy))
+    context.actorOf(Energy.props(valueOfEnergy)(self))
+  }
+
+  private def isInRegion(position: Vector2d): Boolean = {
+    height / Math.abs(height) * position.y > -frontier
   }
+
+  private def isInFrontier(position: Vector2d): Boolean = {
```

Comment:
This is not the correct to do this.
But it's okai for the demo just create an issue related to this or add a `TODO`

Reply:
Lake of time in fact. This issue has already been created.